packages/server/src/stream/service.ts

```diff
@@ -0,0 +1,132 @@
+import { uuid } from '@botpress/messaging-base'
+import clc from 'cli-color'
+import yn from 'yn'
+import { Service } from '../base/service'
+import { ActionSource } from '../base/source'
+import { ChannelService } from '../channels/service'
+import { ClientService } from '../clients/service'
+import { ConduitService } from '../conduits/service'
+import { ConversationEvents } from '../conversations/events'
+import { ConversationService } from '../conversations/service'
+import { HealthEvents } from '../health/events'
+import { HealthService } from '../health/service'
+import { Logger } from '../logger/types'
+import { MappingService } from '../mapping/service'
+import { MessageEvents } from '../messages/events'
+import { MessageService } from '../messages/service'
+import { PostService } from '../post/service'
+import { SocketService } from '../socket/service'
+import { UserEvents } from '../users/events'
+import { UserService } from '../users/service'
+import { WebhookService } from '../webhooks/service'
+
+export class StreamService extends Service {
+  private logger = new Logger('Stream')
+
+  constructor(
+    private posts: PostService,
+    private sockets: SocketService,
+    private channels: ChannelService,
+    private clients: ClientService,
+    private webhooks: WebhookService,
+    private conduits: ConduitService,
+    private health: HealthService,
+    private users: UserService,
+    private conversations: ConversationService,
+    private messages: MessageService,
+    private mapping: MappingService
+  ) {
+    super()
+  }
+
+  async setup() {
+    this.health.events.on(HealthEvents.Registered, async ({ event }) => {
+      const conduit = await this.conduits.get(event.conduitId)
+      const client = await this.clients.getByProviderId(conduit!.providerId)
+      if (!client) {
+        return
+      }
+
+      const channel = this.channels.getById(conduit!.channelId)
+      await this.stream(
+        'health.new',
```

Comment:
I think it would be nice if those events were put in a certain file (client maybe?) so that a user that listens to those events can easily register and does not have to read the code to find the list of available events. What do you think?

Reply:
Yes we could have something similar to telegraf. We provide a middleware to be put in a router 

```ts
this.router.post(
  '/callback',
  messaging.callback()
)
```

and then we can do 

```ts
// messaging is an instance of MessagingClient
messaging.on('health.new', (msg) => { /* do something */ })
messaging.on('message.new', (msg) => { /* do something */  })
```
Alternatively we could do something like that as well. It's not as simple but it scales better  :
```ts
messaging.health.events.on(HealthEvents.New, (msg) => { /* do something */  })
messaging.messages.events.on(MessageEvents.New, (msg) => { /* do something */  })
```

For the socket we could use this approach as well. It could also be useful to use an object oriented approach to the objects sent in the callback. For example :

```ts
messaging.messages.events.on(MessageEvents.New, (msg) => {
  msg.reply(`Hello ${msg.userId}!`)
})

// instead of
messaging.messages.events.on(MessageEvents.New, async (msg) => {
  await messaging.messages.create(msg.conversationId, { type: 'text', text: `Hello ${msg.userId}!`)
})
```